### ISSUE TITLE
Ignore empty log messages (including messages cleared by listeners)

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#5089]
Listeners using the v3 API can modify the logged message, but they cannot remove it altogether. Even if they set the message to an empty string, that empty string is still logged. This adds unnecessary noise to the log file and increases the size of output.xml.

This is easiest to implement by ignoring log messages having an empty string as the value. Not writing them to output.xml is easy and that automatically takes care of removing them from the log file as well. They should not be included in the result model created during execution either, but I actually believe that model doesn't get log messages at all at the moment.

---

**UPDATE:** This change will obviously mean that all empty messages are ignored. Updated the title accordingly.

**Comment by Meet2rohit99** [comment]
@pekkaklarck , i am new to this repo and would like to fix this issue. could you please assign this to me.

**Comment by OukhellouFst** [comment]
@pekkaklarck, I'm currently working with Robotframework to do E2E testing for our application. I'll be glad to fix this simple issue

**Comment by pekkaklarck** [comment]
Anyone is free to create a pull request with the fix and needed tests. Assuming we implement this simply by not writing empty messages to output.xml, the implementation is pretty easy. Creating a test isn't hard either, but requires being familiar with Robot's acceptance test system and designing where to add the test.

Do you @Meet2rohit99 and @OukhellouFst want to agree who tries to create a PR first or do you both want to do it? I'm happy to review PRs either way and can also help otherwise if needed.

**Comment by pekkaklarck** [comment]
Is anyone working with this issue? If not, I can implement it myself as part of other listener enhancements.

**Comment by Aadil-5122** [comment]
@pekkaklarck, I am new to this repo, and working on it.

**Comment by Aadil-5122** [comment]
Here goes the Pull Request:
https://github.com/robotframework/robotframework/pull/5168

**Comment by pekkaklarck** [comment]
The PR mentioned above only worked with messages logged by libraries using `robot.api.logger`, not if a message is cleared by a listener. It's easiest to do the latter by not writing empty messages to the generated output.xml. The aforementioned PR made me thinking that empty messages logged by libraries could be dropped already earlier, though.

**Comment by pekkaklarck** [comment]
I got this implemented and when I run all tests there are some unexpected failures. The failures were due to empty messages logged with the `Log Many` keyword being ignored as well and that's actually not good at all. For example, if I use something like
```
Log Many    ${msg1}    ${msg2}    ${msg3}    ${msg4}
```
there are nowadays always four messages regardless is some of these messages empty or not, but after the change some (or even all) messages could be omitted altogether and mapping actual logged messages to variables would be harder.

The problem explained above is pretty severe and in my opinion benefits of the change are smaller. Possibly we could try to somehow special case `Log Many`, but benefits would anyway be relatively small so that's unlikely to be worth the effort.

I believe it's best to just forget about this altogether. If listeners need to clear a message an leaving an empty message behind is a problem, they can use some canned message explaining that the original message was removed. If that's not enough, a pre-Rebot modifier can remove empty messages after execution.
